File: scala/src/letMap.scala
object letMap {

  def letMap(hof: List[Int] => Unit): Unit = {
    val x = 5
    val list = List.fill(1000000)(7)
    val newList = list.map(y => x+y)
    hof(newList)
  }

  def letMap2(hof: List[Int] => Unit): Unit = {
    val x = 5+5
    val list = List.fill(1000000)(7)
    val newList = list.map(y => x+y)
    hof(newList)
  }
}

File: OCaml/dovs-dev-0.1.0/dovs-dev-0.1.0/benchmarks/letMap.ml
let letMap = 
    let x = 5 in
    let list = List.init 1000000 (fun _ -> 7) in
    let newList = List.map (fun y -> x + y) list in
    List.iter (Printf.printf "%d ") newList

    
let letMap2 = 
    let x = 5+5 in
    let list = List.init 1000000 (fun _ -> 7) in
    let newList = List.map (fun y -> x+y) list in
    List.iter (Printf.printf "%d ") newList